MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java

```diff
@@ -528,34 +529,81 @@ public void actionPerformed(ActionEvent action) {
                 Person spouse = gui.getCampaign().getPerson(UUID.fromString(data[1]));
                 String surnameOption = data[2];
 
-                String selectedSurname = selectedPerson.getName().split(SPACE, 2)[1];
-                String spouseSurname = spouse.getName().split(SPACE, 2)[1];
-                String selectedGivenname = selectedPerson.getName().split(SPACE, 2)[0];
-                String spouseGivenname = spouse.getName().split(SPACE, 2)[0];
+                String[] selectedName = selectedPerson.getName().split(SPACE, 2);
+                String[] spouseName = spouse.getName().split(SPACE,2);
+                String selectedGivenName = "", selectedSurname = "", spouseGivenName = "", spouseSurname = "";
+
+                if (selectedName.length >= 2) {
+                    selectedGivenName = selectedName[0];
+                    selectedSurname = selectedName[1];
```

Comment:
If >= 2 wouldn't the surname be selectedName[selectedName.length - 1] ?

Reply:
You are correct, and the GivenName would be the rest of the name with the removal of the Surname. I'm going to assume a single last name for this PR, and create a RFE to permit multiple last names